### ISSUE TITLE
is this the same as the monitoring part of bbb-exporter?

### ISSUE THREAD

**Issue by mplotsker** [greenstatic/bigbluebutton-monitoring#7]
Can it coexist with the full stack monitoring part of https://github.com/greenstatic/bigbluebutton-exporter?

**Comment by greenstatic** [comment]
Yes, I myself am running such an installation. Both bigbluebutton-monitoring and bigbluebutton-exporter use BBB's API. They do not interfere between each other.

**Comment by mplotsker** [comment]
what is the url to see that page?
your-bbb.com/????

for exporter its your-bbb.com/monitoring

thanks

**Comment by greenstatic** [comment]
bigbluebutton-monitoring is a docker image that hosts the page on port 5000. If you used the provided docker-compose file, you will see that the web server is exposed on the host's networking stack on 127.0.0.1:4000 https://github.com/greenstatic/bigbluebutton-monitoring/blob/b31a52f3eea81f7227bbaad349a1e5d82b1db265/docker-compose.yaml#L7 

You can add this to your nginx config, just like with bigbluebutton-exporter: https://bigbluebutton-exporter.greenstatic.dev/installation/all_in_one_monitoring_stack/#5-configure-nginx

You can use whichever endpoint you would like. Just make sure they don't collide with endpoints BBB uses.
Example:
```yaml
# Reverse proxy for bigbluebutton-monitoring
location /monitoring2/ {
  proxy_pass http://127.0.0.1:4000/;
  include proxy_params;
}
```

Feel free to close the issue if you don't have any other questions.

**Comment by mplotsker** [comment]
this in  /etc/bigbluebutton/nginx/monitoring.nginx from exporter

# BigBlueButton monitoring
location /monitoring/ {
  proxy_pass http://127.0.0.1:3001/;
  include proxy_params;


what do I add to or replace to make this monitoring work?
this?

# Reverse proxy for bigbluebutton-monitoring
location /monitoring2/ {
  proxy_pass http://127.0.0.1:4000/;
  include proxy_params;
}


what about the 4000:5000 in docker-compose.yaml? do i choose one port
or leave both in there?


thank you for the the quick responses.


Michael

**Comment by mplotsker** [comment]
should it look like this?

# BigBlueButton monitoring
location /monitoring/ {
  proxy_pass http://127.0.0.1:3001/;
  include proxy_params;
}

location /monitoring2/ {
 proxy_pass http://127.0.0.1:4000/;
 include proxy_params;
}

 or do i include this line somewhere?

# Reverse proxy for bigbluebutton-monitoring


thanks again

Michael

**Comment by greenstatic** [comment]
It should be like this:
```
# BigBlueButton monitoring
location /monitoring/ {
  proxy_pass http://127.0.0.1:3001/;
  include proxy_params;
}

# Reverse proxy for bigbluebutton-monitoring
location /monitoring2/ {
 proxy_pass http://127.0.0.1:4000/;
 include proxy_params;
}
```

The line `127.0.0.1:4000:5000` means map the socket `127.0.0.1 4000` on the host's networking stack, to port `5000` on the containers networking stack (since the web server in the container, i.e. bigbluebutton-monitoring web server. is listening 
on port 5000).

We could have used port 5000 on the host, but didn't because if I recall correctly, it was used by BBB for some service.

**Comment by mplotsker** [comment]
i think I got it except i do not see any sessions

[image: image.png]
is there some permissions I need in the docker-compose.yaml or secrets.env
file to expose the data? what did I do wrong?

thanks
michael

**Comment by greenstatic** [comment]
The image didn't attach correctly. Try using GitHub's web application and not email reply.

**Comment by mplotsker** [comment]
![Screenshot 2020-12-09 140519](https://user-images.githubusercontent.com/75745665/101675278-9c93ce80-3a27-11eb-9e72-f54dd90a3d18.jpg)

**Comment by greenstatic** [comment]
are you sure you provided the correct `API_BASE_URL` and `API_SECRET`?

Check the logs; go to the directory where docker-compose is located and run `docker-compose logs -f --tail=100`

**Comment by mplotsker** [comment]
do I need to restart bbb? i restarted nginx but not bbb.

Thanks

**Comment by greenstatic** [comment]
No, restarting BBB should not be necessary.

**Comment by greenstatic** [comment]
Nginx clearly works, and the bigbluebutton-monitoring is clearly up. The issue lies in the bbb-monitoring -> bbb api communication. Probably a wrong URL or secret key.

**Comment by greenstatic** [comment]
This is an example URL: `https://bbb.example.com/bigbluebutton/api/` - notice the trailing slash, it's important!

**Comment by mplotsker** [comment]
screwed up the url. fixed it. all is well. can we get the internal and or external meeting id on this dashboard?

thanks
Michael

**Comment by mplotsker** [comment]
thanks again for your speedy help.

Have a great day
Michael

**Comment by greenstatic** [comment]
No problem. Glad you got it fixed.

> Can we get the internal and or external meeting id on this dashboard

Definitely possible, we would only need to send the data to the frontend since iirc it is sent already in the payload from BBB's API. I would gladly accept a PR. :)

**Comment by rexon07** [comment]
Hi @greenstatic 

does this app pull the same details if configured with  scalelite  secrets ?

**Comment by greenstatic** [comment]
@rexon07 I don't have a scalelite configuration to test this, but user's of bigbluebutton-exporter report that it works. bbb-exporter and bbb-monitoring share the same API library to communicate with BBB (albeit bbb-monitoring is using an older version). Come back with your findings :) 

BTW, open a separate issue next time
